python/cuml/test/test_one_hot_encoder.py

```diff
@@ -305,3 +305,35 @@ def test_onehot_categories_shape_mismatch(as_array):
 
     with pytest.raises(ValueError):
         OneHotEncoder(categories=categories, sparse=False).fit(X)
+
+
+@pytest.mark.xfail(strict=True, raises=TypeError)
+def test_onehot_category_overflow():
+    # See this for reasoning: https://github.com/rapidsai/cuml/issues/2690
+    example_df = DataFrame()
+    example_df["high_cardinality_column"] = cp.linspace(0, 255, 256)
```

Comment:
should probably have an example that requires uint32 too. (Will that be unbearably slow?)

Reply:
I added a few more tests, but uint32 causes an out of memory error, so I dont think its wise to add this.